Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -418,6 +487,76 @@ extension Channel {
+    ///     This type must match the `InboundOut` type of the final handler added to the stream channel by the `inboundStreamInitializer`
+    ///     or `HTTP2Frame.Payload` if there are none.
+    ///     - streamOutboundType: The ``NIOAsyncChannel/outboundWriter`` message type for inbound stream channels.
+    ///     This type must match the `OutboundIn` type of the final handler added to the stream channel by the `inboundStreamInitializer`
+    ///     or `HTTP2Frame.Payload` if there are none.
+    ///     - connectionInitializer: A closure that will be called once the `NIOHTTP2Handler` has been added to the pipeline.     
+    ///     - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
+    /// - returns: An `EventLoopFuture` containing the `AsyncStreamMultiplexer` inserted into this pipeline which wraps
+    /// inbound streams as `NIOAsyncChannels` after initialization. The multiplexer can be used to initiate new streams
+    /// and iterate over inbound HTTP/2 stream channels.
+    @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+    @_spi(AsyncChannel)
+    public func configureAsyncHTTP2Pipeline<ConnectionInbound, ConnectionOutbound, StreamInbound, StreamOutbound>(
+        mode: NIOHTTP2Handler.ParserMode,
+        connectionConfiguration: NIOHTTP2Handler.ConnectionConfiguration,
+        streamConfiguration: NIOHTTP2Handler.StreamConfiguration,
+        streamDelegate: NIOHTTP2StreamDelegate? = nil,
+        connectionInboundType: ConnectionInbound.Type,
+        connectionOutboundType: ConnectionOutbound.Type,
+        streamInboundType: StreamInbound.Type,
+        streamOutboundType: StreamOutbound.Type,
+        connectionInitializer: @escaping NIOHTTP2Handler.ConnectionInitializer,
+        inboundStreamInitializer: @escaping NIOHTTP2Handler.StreamInitializer
+    ) throws -> EventLoopFuture<(
+        NIOAsyncChannel<ConnectionInbound, ConnectionOutbound>,
+        NIOHTTP2Handler.AsyncStreamMultiplexer<NIOAsyncChannel<StreamInbound, StreamOutbound>>
+    )> {
+        return try self.configureAsyncHTTP2Pipeline(
+            mode: mode,
+            connectionConfiguration: connectionConfiguration,
+            streamConfiguration: streamConfiguration,
+            streamDelegate: streamDelegate,
+            streamInboundType: streamInboundType,
+            streamOutboundType: streamOutboundType,
+            inboundStreamInitializer: inboundStreamInitializer
+        ).flatMap { multiplexer in
+            return connectionInitializer(self).flatMapThrowing { _ in
+                return try NIOAsyncChannel(
+                    synchronouslyWrapping: self,
+                    inboundType: ConnectionInbound.self,
+                    outboundType: ConnectionOutbound.self
+                )
+            }.map { connectionAsyncChannel in
+                return (connectionAsyncChannel, multiplexer)
```

Comment:
we can get rid of the `map`:

```suggestion
                let connectionChannel = try NIOAsyncChannel(
                    synchronouslyWrapping: self,
                    inboundType: ConnectionInbound.self,
                    outboundType: ConnectionOutbound.self
                )

                return (connectionChannel, multiplexer)
```